cli/azd/pkg/messaging/service.go

```diff
@@ -0,0 +1,57 @@
+package messaging
+
+import (
+	"context"
+)
+
+type Service struct {
+	topics map[string]*Topic
+}
+
+func NewService() *Service {
+	return &Service{
+		topics: map[string]*Topic{},
+	}
+}
+
+type contextKey string
+
+const (
+	defaultTopicName string     = "default"
+	topicContextKey  contextKey = "messaging-topic"
+)
+
+func (s *Service) WithTopic(ctx context.Context, topicName string) context.Context {
+	return context.WithValue(ctx, topicContextKey, topicName)
+}
+
+func (s *Service) Send(ctx context.Context, msg *Message) {
+	topic := s.ensureTopic(ctx)
+	topic.Send(ctx, msg)
+}
+
+func (s *Service) Subscribe(ctx context.Context, filter MessageFilter, handler MessageHandler) *Subscription {
+	topic := s.ensureTopic(ctx)
+	return topic.Subscribe(ctx, filter, handler)
+}
```

Comment:
By default the service wrapper will use a `default` topic but additional topics can be used for more complex messaging orchestration requirements.